### DISCUSSION TITLE
REQ: Option for multiple prime towers or material wall selection for multi-material prints

### DISCUSSION THREAD

**Discussion by storagejarett** [OrcaSlicer/OrcaSlicer#5348] [Ideas]
I often print with multiple materials.  An example is using PETG as a support interface to PLA/PLA+

The problem is that the prime tower often splits and has layer separation, ruining the print.   
Below is an example of a the first 12 hours of a 36hr print.  It split multiple times and I had to tape it, glue it, melt it, etc to keep it together and keep the print going.  In past prints, I have had these failures every 5 or so layers. 

Yes, these are extreme methods to hold the petg & pla tower together, but I do not want to restart the print ever time.  

It would be great if we could have more vast prime tower options:  These would be good is there is a switch in material type on almost every layer.  
- Option 1: Option to have the outer perimeter walls always be 1 of the materials. This would encapsulate the tower and make it stronger and less prone to layer separation.  For example, if Material A (PLA) was the primary filament of the print, it would always be an outer wall of X thickness on the prime tower.  The other material (PETG/PVA/other), would print the prime pattern inside of it. 
- Option 2:  Multiple prime towers, 1 per material.  

Thoughts?

![multimaterial-failures](https://github.com/SoftFever/OrcaSlicer/assets/1211981/2e3818b2-0ef3-4c52-a803-7696e01adb99)

**Comment by vgdh** [comment]
It will force filament change for prime tower print even though there is no usage of that filament in current layer. And this significantly increase print time. But may be it can be useful for you if you willing to wait additional time.

**Reply by storagejarett** [reply]
Yes, that is the downside of using an option like #2.  Though, in my case, I would be willing to accept that as I already have a material change at almost every layer. 

For option 1, where the tower uses external walls to encapsulate the tower, it would not be a major increase in overall number of changes.   This could be an option under support, just like there is a support/raft interface option.

**Comment by vgdh** [comment]
may be it will be useful to make tower with number walls as much as filament used. 
So every filament has its own wall in the tower. 
When filament isn't used it could be printed with current filament in the extruder. 
That way the walls will be fused very good because the same material for each wall.

**Reply by Anfernee12345** [reply]
Has this been put to work yet or its not even in plans ?
Cause i really mis this feature right now. At least the one where i could print the outer wall of the prime tower with main filament, and then support interface filament would be purged inside.

**Reply by vgdh** [reply]
I don't know. The idea need to be implemented and tested my dev team.
May be they just missed this thread. You can try to create issue of an enhancement proposal. And mention some of dev team in it.
